src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2850,6 +2850,12 @@ private static void GetAllCandidates(Dictionary<TypeWithAnnotations, TypeWithAnn
                     if (!ImplicitConversionExists(source, destination, ref useSiteDiagnostics, conversions.WithNullability(false)))
                     {
                         candidates.Remove(candidate);
+                        if (conversions.IncludeNullability)
+                        {
+                            candidates.Remove(bound);
+                            var newBound = TypeWithAnnotations.Create(bound.Type, bound.NullableAnnotation.MergeNullableAnnotation(candidate.NullableAnnotation, variance));
```

Comment:
>var [](start = 28, length = 3)

Might `candidate` and `bound` differ by nested nullability?

Reply:
Candidates never contains multiple equivalent bounds, because it is a `new Dictionary<TypeWithAnnotations, TypeWithAnnotations>(EqualsIgnoringDynamicTupleNamesAndNullabilityComparer.Instance)` (line 2348)

However, since it is being modified here, I need to pull out the corresponding candidate (which may have been modified on a previous iteration) before adding its top-level nullability.

---
In reply to: [267579705](https://github.com/dotnet/roslyn/pull/34270#discussion_r267579705) [](ancestors = 267579705)